src/test/scala/cognite/spark/v1/AssetHierarchyBuilderTest.scala

```diff
@@ -105,8 +117,80 @@ class AssetHierarchyBuilderTest extends FlatSpec with Matchers with SparkTest {
     errorMessage should not include("daughterSon")
   }
 
+  it should "fail reasonably when parent does not exist" in {
+    val tree = Seq(
+      AssetCreate("testNode1", None, None, Some(testName),Some("testNode1"), None, Some("nonExistentNode-jakdhdslfskgslfuwfvbnvwbqrvotfeds")),
+      AssetCreate("testNode2", None, None, Some(testName),Some("testNode2"), None, Some(""))
+    )
+
+    val ex = intercept[Exception] { ingest(tree) }
+    ex shouldBe an[InvalidNodeReferenceException]
+    ex.getMessage shouldBe "Node 'nonExistentNode-jakdhdslfskgslfuwfvbnvwbqrvotfeds' referenced from 'testNode1' does not exist."
+  }
+
+  it should "fail when subtree is updated into being root" in {
```

Comment:
Did they explain why this isn't allowed? It's something I suppose people might want to do, even if it's strange

Reply:
I don't really know, the API simply does not allow you to specify `setNull` to `parentId`.